docs/Explanation/ADR/adr-004-did.md

```diff
@@ -0,0 +1,208 @@
+
+###### Update
+
+Amends to a DID document are allowed only to:
+
+- controllers of one of the verification method listed in the `authentication` verification relationship
+- controllers of one of the verification method listed in the `authentication` verification relationship of the DID document identified in the the controller of the current DID document 
+
+```
+example: 
+
+giving did did:cash:abc:1111 described by document
+
+{
+    "@context": ...
+    "id": "did:cash:abc:1111",           <--- DID 111 subject
+    "controller": [
+        "did:cash:abc:2222"              <--- Controller for the document describing did:cash:abc:1111 
+    ]
+    "verificationMethods" : [
+        "id": "did:cash:abc:1111#key-1",
+        "type": "EcdsaSecp256k1VerificationKey2019",
+        "controller": "did:cash:abc:1111",   <--- Controller for the verification method identified by did:cash:abc:1111#key-1
+        "blockchainAccountID": "cosmos1g4fdrtxjjvzxvazkmj7zgz634a4hn93nry0p9u"
+    ],
+    "authentication": [
+        "id": "did:cash:abc:1111#key-1"      <--- Authentication relationships for key did:cash:abc:1111#key-1
+    ]
+}
+
+and did:cash:abc:2222
+
+{
+    "@context": ...
+    "id": "did:cash:abc:2222",         <--- DID subject 
+    "verificationMethods" : [
+        "id": "did:cash:abc:2222#key-1",
+        "type": "EcdsaSecp256k1VerificationKey2019",
+        "controller": "did:cash:abc:2222", <--- Controller for the verification method identified by did:cash:abc:1111#key-1
+        "blockchainAccountID": "cosmos1ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj",
+    ],
+    "authentication": [
+        "id": "did:cash:abc:2222#key-1"    <--- Authentication relationships for key did:cash:abc:2222#key-1
+    ] 
+}
+
+according to the rules the DID did:cash:abc:2222 is authorized to amend the DID document for did:cash:abc:1111  
+```
+
+
+The transaction signer address MUST match the verification method `blockchainAccountID` for the address part (ignoring the chain prefix).
+
+> XXX: the consequences of that is that a controller has the same powers of the creator of the did document
+
+###### Deactivate
+
+The deactivation of a DID is obtained by removing all the values for the properties:
+
+- `controller`
+- `verificationMethods` 
+
+the deactivation is allowed following the same rules af for the [update operation](#update) 
+
+> 
+
+#### Metadata
+
+
+##### [DID Document Metadata](https://www.w3.org/TR/did-core/#did-document-metadata)
+
+The implementation for metadata MUST report the following properties for a DID document
+
+- `created`: a [datetime](https://www.w3.org/TR/xmlschema11-2/#dateTime) string of the creation date, that is the utc date associated to the block height when the DID document was submitted the first time
+- `deactivated`: whenever the DID document is [deactivated](#Deactivate) 
+- `versionId`: for the version id we use the hash of the transaction that created/updated the DID
```

Comment:
**nitpick:** these should indicate data types for these fields given that `created` gave a type